source/Halibut/Transport/Protocol/MessageSerializer.cs

```diff
@@ -207,6 +207,14 @@ async Task<T> ReadCompressedMessageAsync<T>(Stream stream, IRewindableBuffer rew
 
                 using var deflatedInMemoryStream = new ReadIntoMemoryBufferStream(decompressedByteCountingStream, readIntoMemoryLimitBytes, OnDispose.LeaveInputStreamOpen);
                 await deflatedInMemoryStream.BufferIntoMemoryFromSourceStreamUntilLimitReached(cancellationToken);
+                
+                // If the end of stream was found and we read nothing from the streams
+                if (stream.WasTheEndOfStreamEncountered && compressedByteCountingStream.BytesRead == 0 && decompressedByteCountingStream.BytesRead == 0)
+                {
+                    // Return null since that is what this would do.
```

Comment:
"since that is what this would do" - Could you make it a bit clearer where this would happen and why we need to short circuit